safe_transaction_service/history/services/notification_service.py

```diff
@@ -204,3 +205,54 @@ def build_reorg_payload(block_number: int) -> ReorgPayload:
         blockNumber=block_number,
         chainId=str(get_chain_id()),
     )
+
+
+class DelegatePayload(TypedDict):
+    type: str
+    safeAddress: Optional[str]
+    delegate: str
+    delegator: str
+    label: str
+    expiryDate: Optional[int]
+
+
+def _build_delegate_payload(
+    event_type: Union[
+        TransactionServiceEventType.NEW_DELEGATE,
+        TransactionServiceEventType.UPDATED_DELEGATE,
+        TransactionServiceEventType.DELETED_DELEGATE,
+    ],
+    instance: SafeContractDelegate,
+) -> DelegatePayload:
+    """
+    Build a delegate payload with the specified event type and SafeContractDelegate instance data.
+
+    :param event_type: The transaction event type, restricted to NEW_DELEGATE, UPDATED_DELEGATE, or DELETED_DELEGATE.
+    :param instance: An instance of SafeContractDelegate.
+    :return: A DelegatePayload dictionary with details about the delegate.
+    """
+    return DelegatePayload(
+        type=event_type.name,
+        safeAddress=instance.safe_contract.address if instance.safe_contract else None,
```

Comment:
```suggestion
        safeAddress=instance.safe_contract_id if instance.safe_contract_id else None,
```

This can save a database lookup